contributingGuides/HYBRID_APP.md

```diff
@@ -0,0 +1,212 @@
+
+In practice a submodule is just a regular git repository, set to a specific commit. However, if you need to push changes to `Mobile-Expensify`, you can do all the same things as on a regular repository like: changing branches, pulling and pushing to a remote repository etc. 
+
+If you change the `Mobile-Expensify` code, it will be seen by `Expensify/App` as a commit hash change that may be pushed to a remote repository. Currently we automatically update the `Mobile-Expensify` commit hash, therefore you don't need to do it manually. In order to avoid confusion you can update your `.git/config` by running the following command: `git config submodule.Mobile-Expensify.ignore all`. This way you won't accidentally change the commit hash for the submodule in your PR.
+## Useful commands
+
+IMPORTANT: Please execute the following commands form the root of the project!
+- `git submodule update --init` 
+	- it initialises the submodule, and pulls the `Mobile-Expensify` code from the commit set in `Expesnify/App`. 
+	- **IMPORTANT**: If you already have the `Mobile-Expensify` code, you don't need to run the command with `--init` flag.
+- `git submodule update` 
+	- it pulls the `Mobile-Expensify` code from the commit set in `Expesnify/App`. 
+	- **very similar to:** `git checkout <COMMIT_HASH>`
+- `git submodule update --remote` 
+	- it pulls the `Mobile-Expensify` code from the newest `main` on the remote repository.
+	- **very similar to:** `git checkout main` 
+
+# Patches
+There is a set of patches, that gets applied only to HybridApp. They are not required by a standalone NewDot build, therefore they have been located in `./Mobile-Expensify/patches`. If you'd like to add a new HybridApp specific patch you can run:
+1. `npx patch-package <PACKAGE_NAME> --patch-dir Mobile-Expensify/patches`
+
+The `patch-package` takes `.patch` files, and applies the diff to `node_modules` after executing `npm install`. It means that if you'd like to build NewDot after HybridApp, it is best to remove `node_modules`, and run `npm run i-standalone` to make sure that HybridApp-specific patches won't be applied.
+# Pro Tips & Troubleshooting
+## General
+
+### My build has just failed. What should I do?
+There are multiple thing that may cause the build to fail. This is a simple checklist that may help you with debugging:
+1. Read the error carefully - sometimes it explains the problem very well
+2. Look for a similar error in the Troubleshooting section
+3. Pull the newest `main`
+4. Check if you have a recent version of the `Mobile-Expensify` submodule
+	- if not, execute `git submodule update`, and rerun the build
+5. Clean the cache by executing `npm run clean` (with proper platform flags)
+If the problem still exists, please share the error on the open source channel
```

Comment:
If the problem still exists, please share the error in the [#expensify-open-source](https://expensify.enterprise.slack.com/archives/C01GTK53T8Q) channel.